doc: amend related to execFile explanation (#32545)
nodejs/node: doc/api/child_process.md

  * `stderr` {string|Buffer}
* Returns: {ChildProcess}

> If the `shell` options is not passed as `false`, then a shell is spawned

Reviewer: ```suggestion
If the `shell` option is not passed as `false`, then a shell is spawned
```

[shell](https://en.wikipedia.org/wiki/List_of_command-line_interpreters))
need to be dealt with accordingly:

> If the `shell` option is passed as `false`, then certain shell behaviours such

Reviewer 2: ```suggestion
If the `shell` option is `false`, then certain shell behaviors such
```

(Our style guide standardizes on US English spellings, so `behaviors` instead of `behaviours`.)

Reviewer 2: The documentation above for the `shell` option says it is a string, but here we are talking about passing a boolean. ????

Author: Oh yes, I got your point. When inferring from the code https://github.com/nodejs/node/blob/6107a8f0931a47c92a96f6ea592a4346342276f0/lib/child_process.js#L141 I can see that there is a type check, so passing boolean also will not create an issue and documentation for the same should also be updated.

Reviewer 3: for exec(), the shell defaults to a strong, for execFile(), the shell defaults to a boolean (`false`). I suspect that `true` is not a valid value for options.shell, but have not verified (its definitely not mentioned in the docs).

Author: `true` trigger the usage of default parameters, as `false` is a boolean false. Hence the condition for triggering default values is being used by this `true` value, [link](https://github.com/nodejs/node/blob/e080f223b5b21c61806864e4a8f85ccd6fe4b089/lib/child_process.js#L480).

> If the `shell` option is passed as `false`, then certain shell behaviours such

Reviewer 3: line 174 above isn't correct anymore, its only sometimes true that a shell is  spawned.

The `child_process.execFile()` function is similar to [`child_process.exec()`][]
except that it does not spawn a shell by default. Rather, the specified
executable `file` is spawned directly as a new process making it slightly more

> efficient than [`child_process.exec()`][]. Also `child_process.execFile()` takes

Reviewer 2: I think this sentence should be deleted, but if we keep it, let's add in the missing comma:

```suggestion
efficient than [`child_process.exec()`][]. Also, `child_process.execFile()` takes
```

> efficient than [`child_process.exec()`][]. Also, `child_process.execFile()` takes

Reviewer 3: The efficiency is less of an issue than that shells do metacharacter expansion, so no shell will not.

an array of strings as an argument while the other takes a single command

> string.

Reviewer 2: I'm not sure we need to put this information into prose like this. It's in the function signature.

Reviewer 3: I'm not sure this is the best way to express it, but I do think we have to say something about how exec and execFile differ. If we had to do this again, both functions would *not* exist, there would be only `exec()`, since they now differ only in two things: the default for options.shell, the cmd vs arvgv signature. I think its worth calling this out, because its confusing, and so that people don't have to do a line by line comparison of the two functions to see that those are the only two differences - we can just say it straight up. It probably is not possible to refactor this here, but this is one reason I dislike cut-n-paste in docs as much as in code: if two functions link to the same core docs, its instantly clear that they share the same core behaviour, with cut-n-paste, it takes very careful reading. From a historical perspective, the exec and execFile docs weren't literally copied from each other, its just they got mutual feature envy, and eventually every feature one had was also added to the other so it appears as if the docs were copied.  Its possible that the child_process docs could become much more readable by compressing the multiple copies of the arg docs into core sections, linked to by the APIs that expose that behaviour, but its also possible it would make them more confusing.

> string.

Reviewer 2: I think this sentence should be deleted, but if we're going to keep it, let's avoid ambiguity:

```suggestion
an array of strings as an argument whereas `child_process.exec()` takes
 a single command string.
```
